Formula/smartdns.rb

```diff
@@ -0,0 +1,43 @@
+class Smartdns < Formula
+  desc "Rule-based DNS server for fast IP resolution, DoT/DoH/DoQ supported"
+  homepage "https://github.com/mokeyish/smartdns-rs"
+  url "https://github.com/mokeyish/smartdns-rs/archive/refs/tags/0.3.1.tar.gz"
+  sha256 "3a11e1b1c6ffc33d1120b6cec516d19a90375be283c114771cf950e2b364215d"
+  license "GPL-3.0-only"
+  head "https://github.com/mokeyish/smartdns-rs.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "bind" => :test # for `dig`
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+    sbin.install bin/"smartdns"
+    pkgetc.install "etc/smartdns/smartdns.conf"
+  end
+
+  service do
+    run [opt_sbin/"smartdns", "run", "-c", pkgetc/"smartdns.conf"]
```

Comment:
```suggestion
    run [opt_sbin/"smartdns", "run", "-c", etc/"smartdns/smartdns.conf"]
```
Unfortunately `pkgetc` does not work inside `service` blocks. (We should fix that.)